cpp/include/raft/core/math.hpp

```diff
@@ -102,7 +102,13 @@ template <typename T>
 RAFT_INLINE_FUNCTION auto asin(T x)
 {
 #ifdef __CUDA_ARCH__
-  return ::asin(x);
+  if constexpr (std::is_same<T, __half>::value) {
```

Comment:
do we need half support for the `asin` function?   

I'm wondering if we should either remove the half support for this function, or add half support for all the other trigonometric functions in this file

Reply:
Yeah, maybe we can. As of now, it's a kind of trade-off. I understand your meaning; some distance algorithms need this as supporting half; removing it will cause a compilation error. If we bring half everything, that could be ideal, but the workload can be out of control..